.github/workflows/rust-build-and-test.yml

```diff
@@ -97,6 +100,57 @@ jobs:
         echo "PROTOC=/usr/local/bin/protoc" >> $GITHUB_ENV
         echo "PROTOC_INCLUDE=/usr/local/include" >> $GITHUB_ENV
     - name: Build
-      run: make build
+      run: make build # Update to push an image, look at the cd-to-infra.yml file for an example.
+    -
+      name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    -
+      name: Login to Public ECR
+      uses: docker/login-action@v2
+      with:
+        registry: public.ecr.aws
+        username: ${{ env.AWS_ACCESS_KEY_ID }}
+        password: ${{ env.AWS_SECRET_ACCESS_KEY }}
+      env:
+        AWS_REGION: us-east-1
+    -
+      name: Build and Publish
+      id: build
+      run: |
+        SHA_TAG=$(echo ${{ github.SHA }} | head -c 12)
+        DEPLOY_TAG=$SHA_TAG
+        COMMIT_MESSAGE=$(git log -1 --pretty=%B)
+        if [[ $(echo "$COMMIT_MESSAGE" | grep 'chore: version v') ]]; then
+          RELEASE_TAG=$(cargo metadata --format-version=1 --no-deps | jq '.packages[0].version' | tr -d '"')
+          # Use the release tag to deploy, if one is available.
+          DEPLOY_TAG=$RELEASE_TAG
+        fi
+        make SHA="${{ github.SHA }}" SHA_TAG="$SHA_TAG" RELEASE_TAG="$RELEASE_TAG" publish-docker
+        echo "Deploy tag:"
+        echo ${DEPLOY_TAG}
+        echo "deploy_tag=${DEPLOY_TAG}" >> $GITHUB_OUTPUT
 
+  correctness-tests:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+    - uses: actions/checkout@v3
+    - name: Get Network template and update
+      env:
+        RUST_CERAMIC_IMAGE: ${{ needs.build.outputs.deploy_tag }}
+      run: |
+        sudo apt-get update && apt-get install -y curl yq
+        curl -LO https://github.com/3box/ceramic-tests/releases/download/v0.2.0/hermetic-driver-x86_64-linux
+        mv hermetic-driver-x86_64-linux ./bin/hermetic-driver
+        chmod +x ./bin/hermetic-driver
+        curl -LO https://raw.githubusercontent.com/3box/ceramic-tests/main/networks/basic-rust.yaml
+        yq e '.spec.ceramic[0].ipfs.rust.image = strenv(RUST_CERAMIC_IMAGE)' basic-rust.yaml
+        cat basic-rust.yaml
+    - name: Run correctness-tests
+      run: |
+        ./bin/hermetic-driver test \
```

Comment:
Do k8s creds need to be injected for this to work?

Reply:
Added!